data-protocols/ids/ids-api-multipart-dispatcher-v1/src/main/java/org/eclipse/dataspaceconnector/ids/api/multipart/dispatcher/sender/IdsMultipartSender.java

```diff
@@ -99,8 +100,10 @@ private static URI createConnectorIdUri(String connectorId) {
      */
     @Override
     public CompletableFuture<R> send(M request, MessageContext context) {
+        var remoteConnectorAddress = retrieveRemoteConnectorAddress(request);
+
         // Get Dynamic Attribute Token
-        var tokenResult = identityService.obtainClientCredentials(TOKEN_SCOPE);
+        var tokenResult = identityService.obtainClientCredentials(TOKEN_SCOPE, remoteConnectorAddress);
```

Comment:
why do we need the remoteConnector address for obtaining the Client Credential, this could be done in the method... In this way every one who implements the identity service has this string unused in it or what do you think?

Reply:
The `remoteConnectorAddress` will be the JWT `audience` claim, so that the service receiving the token cannot reuse it to impersonate its caller at another connector.

Reply:
@maximilianschmidt-zf please check out the recent push I made, with a Parameter Object. This way the prototype is not "polluted" anymore